### PR TITLE
Expose the loss function and optimizer params via model object constructors

### LINKED ISSUE #73: Loss function in CNNClassifier
Hi, 

Thanks for the great work! I was looking at the implementation of `CNNClassifier` and noticed that the loss function used for this model appears to be [MSE](https://github.com/sktime/sktime-dl/blob/feb42cf327371531c56c839c9dc24e786c0a34c4/sktime_dl/deeplearning/cnn/_classifier.py#L93). Is there any specific reason for this choice of the loss function? Other models seem to use cross-entropy which should potentially work better for classification problems.

### PR DESCRIPTION
**Pull request by vanyalzr**
#### Reference Issues/PRs

Fixes #73. 

#### What does this implement/fix? Explain your changes.

- changed default loss function of CNNClassifier to `categorical_crossentropy`
- exposed the loss function and optimizer parameters in model constructors so that the users can e.g. tune the initial learning rate of the model through the package API

### CODE REVIEW

**Comment by mloning** [comment]
Thanks @vanyalzr - this is much appreciated! 

We're currently facing some issues with sktime-dl and need to update it to reflect the latest changes in base sktime. This shouldn't be too complicated, but we currently don't have anyone with the time to work on it.

**Comment by vanyalzr** [comment]
Sure, no problem, this PR could be merged whenever the package/CI is stabilized. 

I took the liberty to change the loss function for CNNClassifier to categorical cross-entropy. I think the implementation with MSE follows [the one from here](https://github.com/hfawaz/dl-4-tsc). I ran CNNClassifier on UCR datasets for both loss functions and the resulting difference is centered at zero, I wouldn't say either loss function is better (see histogram over UCR datasets below). Cross-entropy makes more sense for classification in general, though.

drawing

**Comment by TonyBagnall** [comment]
closing this prior to update, feel free to open a new issue after
